### PR TITLE
Bugfix: fix broken of download models from modelscope

### LINKED ISSUE #4362: [Bug]: v0.4.1 VLLM_USE_MODELSCOPE not working
### Your current environment

```text
The output of `python collect_env.py`
```

### 🐛 Describe the bug

the related code in the previous version :

`if os.environ.get("VLLM_USE_MODELSCOPE", "False").lower() == "true":`

now the code:
`if VLLM_USE_MODELSCOPE:`
but the `VLLM_USE_MODELSCOPE` is set to string "true" by
```
VLLM_USE_MODELSCOPE = os.environ.get("VLLM_USE_MODELSCOPE",
                                     "False").lower() == "true"
```
so, VLLM_USE_MODELSCOPE will never work

### PR DESCRIPTION
**Pull request by liuyhwangyh**
FIX https://github.com/vllm-project/vllm/issues/4362

### CODE REVIEW

File: tests/entrypoints/test_model_from_modelscope.py
Comment (simon-mo) [inline tests/entrypoints/test_model_from_modelscope.py:1]: This test isn't ran at the moment, and I would prefer this be put inside test_regression if possible

File: vllm/transformers_utils/config.py [hunk 1]
```diff
@@ -24,11 +25,20 @@ def get_config(model: str,
                code_revision: Optional[str] = None,
                rope_scaling: Optional[dict] = None) -> PretrainedConfig:
     try:
-        config = AutoConfig.from_pretrained(
-            model,
-            trust_remote_code=trust_remote_code,
-            revision=revision,
-            code_revision=code_revision)
+        if VLLM_USE_MODELSCOPE:
```
Comment (simon-mo) [inline vllm/transformers_utils/config.py:28]: can you use the following pattern? 

```python
if VLLM_USE_MODELSCOPE:
    from modelscope import AutoConfig
    auto_cls = AutoConfig
else: 
    from transformers import AutoConfig
    auto_cls = AutoConfig

config = AutoConfig...

Comment (liuyhwangyh) [reply vllm/transformers_utils/config.py:28]: Updated

**Review by simon-mo — APPROVED** [review]

**Comment by simon-mo** [comment]
Lint failed https://github.com/vllm-project/vllm/actions/runs/9380186676/job/25826653408?pr=5233

**Comment by liuyhwangyh** [comment]
> Lint failed https://github.com/vllm-project/vllm/actions/runs/9380186676/job/25826653408?pr=5233

Fixed

**Comment by JerryKwan** [comment]
But it seems does not process VLLM_USE_MODELSCOPE in the following logic
https://github.com/vllm-project/vllm/blob/b74a12580022039fc15fb54c7066cc13fdc96220/vllm/transformers_utils/config.py#L76-L80

What will happen if the model type is chatglm?

**Pull request merged**